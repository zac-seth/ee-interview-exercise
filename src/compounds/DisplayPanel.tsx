import { FC } from 'react';

import { PrimaryNumberDisplay, SecondaryNumberDisplay } from 'elements';

import { MultiLineDisplayPanel } from './MultiLineDisplayPanel.style';

export type DisplayPanelProps = {
  primary: string;
  secondary: string;
};

export const DisplayPanel: FC<DisplayPanelProps> = ({ primary, secondary }) => (
  <MultiLineDisplayPanel data-testid="multi-line-display-panel">
    <PrimaryNumberDisplay value={primary} />
    <SecondaryNumberDisplay value={secondary} />
  </MultiLineDisplayPanel>
);
