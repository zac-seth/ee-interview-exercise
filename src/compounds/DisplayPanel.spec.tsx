import { render, screen } from "@testing-library/react";

import { DisplayPanel } from "./DisplayPanel";

describe(`DisplayPanel`, () => {
  it(`Should render the display panel`, () => {
    render(<DisplayPanel primary="1000" secondary="100 + 250 - 30" />);

    expect(screen.getByTestId(`multi-line-display-panel`)).toMatchSnapshot();
  });
});
