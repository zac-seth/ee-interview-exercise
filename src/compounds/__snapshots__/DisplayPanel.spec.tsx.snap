// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayPanel Should render the display panel 1`] = `
.c1 {
  text-shadow: 0 0 5px #FFF,0 0 15px #B6FF00;
  color: #FFF;
  margin: 0;
  padding: 0;
  font-size: 4rem;
  line-height: 4rem;
}

.c2 {
  text-shadow: 0 0 5px #FFF,0 0 15px #B6FF00;
  color: #FFF;
  margin: 0;
  padding: 0;
  font-size: 3.2rem;
  line-height: 3.2rem;
}

.c0 {
  text-align: right;
  color: #FFF;
  padding: 1rem;
}

<section
  class="c0"
  data-testid="multi-line-display-panel"
>
  <p
    class="c1"
    mode="primary"
  >
    1000
  </p>
  <p
    class="c2"
    mode="secondary"
  >
    100 + 250 - 30
  </p>
</section>
`;
